### PR TITLE
Restore loop fix

### LINKED ISSUE #32: Re-restore fail upon database online  check
This issue should not be encountered in normal use. However if an active database is not in the "online" state & is not being restored with this tool, in theory the code would break. This is because the code checks all active databases state upon completion and tries to restore any that aren't online.

In reference to this code block:

https://github.com/MonitoringSolutionsInc/DatabaseBackupTool/blob/542a4cca7266b3db0c10402cbb7a3746c6ff6db1/DatabaseBackupTool/Restore.cs#L223-L292

### PR DESCRIPTION
**Pull request by SamuraiTheNinja**
### Fixes #32 
<br>
This patch makes sure that all the databases that it checks are in BOTH the list of files AND the list of databases.
I also re arranged a little bit of stuff inside that method.

## We need to make sure we upload a new release when this is tested

### CODE REVIEW

**Comment by SamuraiTheNinja** [comment]
# Never mind, working on an action to do automatic releases now

**Comment by SamuraiTheNinja** [comment]
# Got it. 
### Once this is approved (assuming you agree with the changes) and merged the action will publish a release:
* it will go from 1.4.0 -> 1.4.1 (given no labels)
* it will go from 1.4.0 -> 1.5.0 (given 'minor change' label)
* it will go from 1.4.0 -> 2.0.0 (given 'major change' label)
* will not publish a release (given 'no code changes' label)
* NOTE: 'version change' label cannot work here, since it grabs current version from the latest tag

**Review by JazzGlobal — APPROVED** [review]
Fixes issue #32

**Pull request merged**
